src/backend/parser/parse_relation.c

```diff
@@ -592,6 +592,29 @@ scanRTEForColumn(ParseState *pstate, RangeTblEntry *rte, char *colname,
 	 */
 	if (rte->rtekind == RTE_RELATION)
 	{
+		/* AO tables by nature do not have special system columns like 
```

Comment:
move this comment after getting attnum and before checking it

Reply:
Purpose of getting attnum is checking it, so both things can be considered as a single logical operation. Moreover, this check was the reason why getting attnum was moved here. The fact that attnum used below is just a coincidence. That's why comment is here.